project/src/main/world/creature/creature-loader.gd

```diff
@@ -80,11 +80,26 @@ var _dna_alternatives := DnaAlternatives.new()
 ## Parameters:
 ## 	'include_secondary_creatures': If 'true' the function has a chance to return a creature from a library of
 ## 		predefined creatures instead of a randomly generated one.
+##
+## 	'creature_type': (Optional) The required creature type. If specified, creatures will be skipped in the
+## 		secondary queue until one conforms to the specified type.
 func random_def(include_secondary_creatures: bool = false, creature_type: int = Creatures.Type.DEFAULT) -> CreatureDef:
 	var result: CreatureDef
-	if include_secondary_creatures and PlayerData.creature_queue.has_secondary_creature() and randf() < 0.2:
-		result = PlayerData.creature_queue.pop_secondary_creature()
-	else:
+	if include_secondary_creatures \
+			and PlayerData.creature_queue.has_secondary_creature() \
+			and randf() < 0.2:
+		
+		for i in range(10):
```

Comment:
10 is sort of a random number, leave a comment so you can remember why you picked this number

Reply:
Left a comment, `If we check too few creatures...`